src/Sentry/BaggageHeader.cs

```diff
@@ -0,0 +1,150 @@
+        /// </returns>
+        public static BaggageHeader? TryParse(string baggage, bool onlySentry = false)
+        {
+            // Example from W3C baggage spec:
+            // "key1=value1;property1;property2, key2 = value2, key3=value3; propertyKey=propertyValue"
+
+            var items = baggage.Split(',', StringSplitOptions.RemoveEmptyEntries);
+            var members = new List<KeyValuePair<string, string>>(items.Length);
+
+            foreach (var item in items)
+            {
+                // Per baggage spec, the value may contain = characters, so limit the split to 2 parts.
+                var parts = item.Split('=', 2);
+                if (parts.Length != 2)
+                {
+                    Logger?.LogWarning(
+                        "The baggage header has an item without a '=' separator, and it will be discarded. " +
+                        "The item is: \"{0}\"", item);
+                    continue;
+                }
+
+                var key = parts[0].Trim();
+                if (key.Length == 0)
+                {
+                    Logger?.LogWarning(
+                        "The baggage header has an item with an empty key, and it will be discarded. " +
+                        "The item is: \"{0}\"", item);
+                    continue;
+                }
+
+                var value = parts[1].Trim();
+                if (value.Length == 0)
+                {
+                    Logger?.LogWarning(
+                        "The baggage header has an item with an empty value, and it will be discarded. " +
+                        "The item is: \"{0}\"", item);
+                    continue;
+                }
+
+                if (!onlySentry || key.StartsWith(SentryKeyPrefix))
+                {
+                    // Values are percent-encoded.  Decode them before storing.
+                    members.Add(key, Uri.UnescapeDataString(value));
+                }
+            }
+
+            return members.Count == 0 ? null : new BaggageHeader(members);
+        }
+
+        public static BaggageHeader Create(
+            IEnumerable<KeyValuePair<string, string>> items,
+            bool useSentryPrefix = false)
+        {
+            var members = items.Where(member => IsValidKey(member.Key));
+
+            if (useSentryPrefix)
+            {
+                members = members.Select(kvp => new KeyValuePair<string, string>(SentryKeyPrefix + kvp.Key, kvp.Value));
+            }
+
+            return new BaggageHeader(members);
+        }
+
+        public static BaggageHeader Merge(IEnumerable<BaggageHeader> baggageHeaders) =>
+            new(baggageHeaders.SelectMany(x => x.Members));
+
+        private static bool IsValidKey(string key)
+        {
+            if (key.Length == 0)
+            {
+                return false;
+            }
+
+            // The rules are the same as for HTTP headers.
+            // TODO: Is this public somewhere in .NET we can just call?
```

Comment:
Do we know that already?

Reply:
I could only find it internally (on the .NET source link given).  Not on any publicly available method.